src/modelmanager.cpp

```diff
@@ -1343,7 +1343,7 @@ Status ModelManager::readAvailableVersions(std::shared_ptr<FileSystem>& fs, cons
 Status ModelManager::addModelVersions(std::shared_ptr<ovms::Model>& model, std::shared_ptr<FileSystem>& fs, ModelConfig& config, std::shared_ptr<model_versions_t>& versionsToStart, std::shared_ptr<model_versions_t> versionsFailed) {
     Status status = StatusCode::OK;
     try {
-        status = model->addVersions(versionsToStart, config, fs, *ieCore, versionsFailed, this->metricRegistry, &this->metricConfig);
```

Comment:
Replace std::shared_ptr<model_versions_t> versionsFailed)  with
std::shared_ptr<model_versions_t>& versionsFailed, or even better just with
model_versions_t& versionsFailed

Reply:
The same is true for reloadModelVersions